sample-programs/erc20/erc20.scm

```diff
@@ -0,0 +1,129 @@
+;;  copyright: (c) 2013-2018 by Blockstack PBC, a public benefit corporation.
+
+;;  This file is part of Blockstack.
+
+;;  Blockstack is free software. You may redistribute or modify
+;;  it under the terms of the GNU General Public License as published by
+;;  the Free Software Foundation, either version 3 of the License or
+;;  (at your option) any later version.
+
+;;  Blockstack is distributed in the hope that it will be useful,
+;;  but WITHOUT ANY WARRANTY, including without the implied warranty of
+;;  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+;;  GNU General Public License for more details.
+
+;;  You should have received a copy of the GNU General Public License
+;;  along with Blockstack. If not, see <http://www.gnu.org/licenses/>.
+
+;; Storage
+(define-map balances 
+  ((owner principal)) 
+  ((balance int)))
+(define-map allowances 
+  ((spender principal) (owner principal)) 
+  ((allowance int)))
+(define total-supply 0)

```

Comment:
`total-supply` isn't used anywhere.  Shouldn't it be used to prevent someone from minting too many tokens?

Reply:
Introducing a cap in the total-supply would make this implementation opinionated.